oracle/src/poseidon.rs

```diff
@@ -34,13 +40,16 @@ impl SpongeConstants for PlonkSpongeConstantsLegacy {
 #[derive(Clone)]
 pub struct PlonkSpongeConstantsKimchi {}
 
-impl SpongeConstants for PlonkSpongeConstantsKimchi {
-    const ROUNDS_FULL: usize = 55;
-    const ROUNDS_PARTIAL: usize = 0;
-    const HALF_ROUNDS_FULL: usize = 0;
+impl SpongeParams for PlonkSpongeConstantsKimchi {
     const SPONGE_CAPACITY: usize = 1;
     const SPONGE_WIDTH: usize = 3;
     const SPONGE_RATE: usize = 2;
+}
+
+impl PermutationParams for PlonkSpongeConstantsKimchi {
```

Comment:
it feels a bit weird to me that both `SpongeParams` and `PermutationParams` are implemented on `PlonkSpongeConstantsKimchi`. I would think that you should be able to:

* instantiate a specific permutation via some parameters
* instantiate a sponge using sponge parameters AND a specific permutation. 

Does that make sense? I'm not too sure how this can be cleanly done. Perhaps, `ArithmeticSponge` should also take a `P: Permutation` with:

```rust
trait Permutation {
    const ROUNDS_FULL: usize;
    const ROUNDS_PARTIAL: usize;
    const HALF_ROUNDS_FULL: usize;
    const SPONGE_BOX: u32;
    const FULL_MDS: bool;
    const INITIAL_ARK: bool;
 
    fn poseidon_block_cipher(params: &ArithmeticSpongeParams<F>, state: &mut Vec<F>) {
      // default implementation in the trait
    }
}

struct PermutationKimchi;

impl Permutation for PermutationKimchi {
    const ROUNDS_FULL: usize = 55;
    const ROUNDS_PARTIAL: usize = 0;
    const HALF_ROUNDS_FULL: usize = 0;
    const SPONGE_BOX: u32 = 7;
    const FULL_MDS: bool = true;
    const INITIAL_ARK: bool = false;
}
```

Reply:
@mimoo I'm in the process of replying on your issue, but this proposal in particular seems to make the code more complex for no benefit.

Reply:
I agree for the current version, I'm still curious if we can land on something that looks good (even if we don't merge it quickly, to make sure there's no other issues in snarkyjs & co). 

@frisitano, I don't want to make you lose your time here, with the possibility of this not being landed (we've had issues with complex changes in oracle in the past). I'm still interested to see where this could go, but happy to suggest another task if you want to do something that'll be merged for sure :)